include/boost/compute/detail/parameter_cache.hpp

```diff
@@ -117,9 +118,16 @@ class parameter_cache : boost::noncopyable
     static std::string version_string()
     {
         char buf[32];
-        std::snprintf(buf, sizeof(buf), "%d.%d.%d", BOOST_COMPUTE_VERSION_MAJOR,
-                                                    BOOST_COMPUTE_VERSION_MINOR,
-                                                    BOOST_COMPUTE_VERSION_PATCH);
+        // snprintf is in Visual Studio since Visual Studio 2015 (_MSC_VER == 1900)
+        #if defined (_MSC_VER) && _MSC_VER < 1900
+            #define DETAIL_SNPRINTF sprintf_s
+        #else
+            #define DETAIL_SNPRINTF std::snprintf
+        #endif
+        DETAIL_SNPRINTF(buf, sizeof(buf), "%d.%d.%d", BOOST_COMPUTE_VERSION_MAJOR,
+                                                      BOOST_COMPUTE_VERSION_MINOR,
+                                                      BOOST_COMPUTE_VERSION_PATCH);
```

Comment:
I wonder if there's any utilities somewhere in Boost for doing this platform-specific stuff..